pkg/cmd/server/server.go

```diff
@@ -0,0 +1,172 @@
+package server
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apiserver"
+
+	//"k8s.io/kubernetes/pkg/api"
+	//"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/genericapiserver"
+	genericserveroptions "k8s.io/kubernetes/pkg/genericapiserver/options"
+)
+
+// ServerOptions contains the aggregation of configuration structs for
+// the service-catalog server. The theory here is that any future user
+// of this server will be able to use this options object as a sub
+// options of it's own.
+type ServerOptions struct {
+	// the runtime configuration of our server
+	GenericServerRunOptions *genericserveroptions.ServerRunOptions
+	// the https configuration. certs, etc
+	SecureServingOptions *genericserveroptions.SecureServingOptions
+	// storage with etcd
+	EtcdOptions *genericserveroptions.EtcdOptions
+	// authn
+	AuthenticationOptions *genericserveroptions.DelegatingAuthenticationOptions
+	// authz
+	AuthorizationOptions *genericserveroptions.DelegatingAuthorizationOptions
+}
+
+// I made this up to match some existing paths. I am not sure if there
+// are any restrictions on the format or structure beyond text
+// separated by slashes.
+const etcdPathPrefix = "/k8s.io/incubator/service-catalog"
+
+// I made this up. Maybe we'll need it.
+const GroupName = "service-catalog.incubator.k8s.io"
+
+func NewCommandServer(out io.Writer) *cobra.Command {
```

Comment:
Its a minor thing, and only change it if you make other changes but the use of the word "Command" here could be confusing - as in "what's a command server?" vs just a normal server?  I'd prefer to just drop the word.